keyform/views.py

```diff
@@ -13,6 +13,34 @@ class HomeView(LoginRequiredMixin, ListView):
     model = Request
     template_name = "keyform/home.html"
 
+    def get_context_data(self):
+        context = super(HomeView, self).get_context_data()
+        context["request_types"] = Request.REQUEST_TYPES
+        context["status_types"] = Request.STATUS_TYPES
+        return context
+
+
+class SearchView(LoginRequiredMixin, ListView):
+    template_name = "keyform/home.html"
+
+    def get_queryset(self):
+        qset = Request.objects.all()
+        valid = ['amt_recieved', 'bpn', 'building', 'building_id', 'charge_amount', 'charged_on_rcr',
+                 'comment', 'created_timestamp', 'id', 'keydata__room_number', 'keydata__core_number', 'keydata__key_number',
+                 'payment_method', 'reason_for_request', 'staff', 'staff_id', 'status', 'student_name']
+
+        for item, value in self.request.GET.items():
+            if str(item) in valid:
+                if value != '':
+                    qset = qset.filter(**{item + '__icontains': value})
```

Comment:
You still need to implement a way to filter over the date range provided by `start_date` and `end_date`